cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -355,7 +355,7 @@ func StartControllers(s *options.CMServer, kubeClient *client.Client, kubeconfig
 
 		if containsResource(resources, "deployments") {
 			glog.Infof("Starting deployment controller")
-			go deployment.NewDeploymentController(clientset.NewForConfigOrDie(restclient.AddUserAgent(kubeconfig, "deployment-controller")), ResyncPeriod(s)).
+			go deployment.NewDeploymentController(podInformer, clientset.NewForConfigOrDie(restclient.AddUserAgent(kubeconfig, "deployment-controller")), ResyncPeriod(s)).
```

Comment:
Do not update the deployment controller to share the pod cache just yet because it mutates its pod cache! I've opened https://github.com/kubernetes/kubernetes/issues/27687